optimum/neuron/modeling_decoder.py

```diff
@@ -124,20 +119,25 @@ def _from_transformers(
             local_files_only=local_files_only,
             force_download=force_download,
             trust_remote_code=trust_remote_code,
-            **model_kwargs,
+            **kwargs,
         )
 
         # Save the model checkpoint in a temporary directory
         checkpoint_dir = TemporaryDirectory()
         save_pretrained_split(model, checkpoint_dir.name)
 
+        # If the sequence_length was not specified, deduce it from the model configuration
+        if sequence_length is None:
+            # Note: for older models, max_position_embeddings is an alias for n_positions
+            sequence_length = config.max_position_embeddings
```

Comment:
May we leverage the [`NormalizedConfig`](https://github.com/huggingface/optimum/blob/v1.13.2/optimum/utils/normalized_config.py#L23), and infer it with something like `self._normalized_config.sequence_length`? This could help avoid some edged cases.

Reply:
I will need some support on that because I don't know how `NormalizedConfig` is used. Let's discuss it next week.